code/game/machinery/big_manipulator.dm

```diff
@@ -107,8 +107,21 @@
 	var/mob/monkey_resolve = monkey_worker?.resolve()
 	if(!isnull(monkey_resolve))
 		monkey_resolve.forceMove(get_turf(monkey_resolve))
-		monkey_resolve.pixel_x = initial(monkey_resolve.pixel_x)
-		monkey_resolve.pixel_y = initial(monkey_resolve.pixel_y)
+
+/obj/machinery/big_manipulator/Exited(atom/movable/gone, direction)
+	if(isnull(monkey_worker))
+		return
+	var/mob/living/carbon/human/species/monkey/poor_monkey = monkey_worker.resolve()
+	if(gone != poor_monkey)
+		return
+	if(!is_type_in_list(poor_monkey, manipulator_hand.vis_contents))
+		return
+	manipulator_hand.vis_contents -= poor_monkey
+	if(manipulate_mode == USE_ITEM_MODE)
+		change_mode()
+	poor_monkey.pixel_x = poor_monkey.base_pixel_x + poor_monkey.body_position_pixel_x_offset
+	poor_monkey.pixel_y = poor_monkey.base_pixel_y + poor_monkey.body_position_pixel_y_offset
```

Comment:
It seems a PR that refactored pixel offsets was merged a couple days after I dropped my review. You can remove the `body_position_pixel_*_offset` vars.
However, now that I think about it, the pixel w and z vars should be used in place of x and y since they don't mess with layering.